src/main/python/systemml/converters.py

```diff
@@ -224,24 +255,45 @@ def convertImageToNumPyArr(im, img_shape=None, add_rotated_images=False, add_mir
         size = None
     if num_channels != 1 and num_channels != 3:
         raise ValueError('Expected the number of channels to be either 1 or 3')
+
+    from PIL import Image
+
     if size is not None:
         im = im.resize(size, Image.LANCZOS)
     expected_mode = 'L' if num_channels == 1 else 'RGB'
     if expected_mode is not im.mode:
         im = im.convert(expected_mode)
+
     def _im2NumPy(im):
         if expected_mode == 'L':
             return np.asarray(im.getdata()).reshape((1, -1))
         else:
-            # (H,W,C) --> (C,H,W) --> (1, C*H*W)
-            return np.asarray(im).transpose(2, 0, 1).reshape((1, -1))
+            im = (np.array(im).astype(np.float))
```

Comment:
Means are defined in float values. Image values are in integer (0-255), without conversion precision will be lost.